model-optimizer/extensions/ops/non_max_suppression.py

```diff
@@ -57,31 +55,56 @@ def backend_attrs(self):
 
     @staticmethod
     def infer(node: Node):
+        num_of_inputs = len(node.in_ports())
+        opset = node.get_opset()
+        max_num_of_inputs = 6 if opset == 'opset5' else 5
+        input_msg_fmt = 'NonMaxSuppression node {} from {} must have from 2 to {} inputs'
+        inputs_msg = input_msg_fmt.format(node.soft_get('name', node.id), opset, max_num_of_inputs)
+        assert 2 <= num_of_inputs <= max_num_of_inputs, inputs_msg
+
         boxes_shape = node.in_port(0).data.get_shape()
         assert boxes_shape is not None, 'The shape of tensor with boxes is not defined'
         scores_shape = node.in_port(1).data.get_shape()
         assert scores_shape is not None, 'The shape of tensor with scores is not defined'
         assert len(boxes_shape) == 3, 'Length of tensors with boxes must be equal to 3'
         assert len(scores_shape) == 3, 'Length of tensors with scores must be equal to 3'
 
-        max_output_boxes_per_class = node.in_port(2).data.get_value()
-        if max_output_boxes_per_class is None:
+        if num_of_inputs >= 3:
+            max_output_boxes_per_class = node.in_port(2).data.get_value()
+        else:
+            max_output_boxes_per_class = 0
+
+        if not max_output_boxes_per_class:
             log.info('Set default "max_output_boxes_per_class" for node {} to number of boxes'.format(node.name))
             max_output_boxes_per_class = boxes_shape[1]
 
         num_classes = scores_shape[1]
         num_input_boxes = boxes_shape[1]
         assert scores_shape[2] == num_input_boxes, 'Number of boxes mismatch'
 
-        if node.get_opset() == 'opset4':
+        if node.get_opset() in ['opset4', 'opset5']:
             max_number_of_boxes = min(num_input_boxes, max_output_boxes_per_class) * boxes_shape[0] * num_classes
         else:
             max_number_of_boxes = min(num_input_boxes, boxes_shape[0] * max_output_boxes_per_class * num_classes)
         node.out_port(0).data.set_shape(int64_array([max_number_of_boxes, 3]))
 
+        num_of_outputs = len([port for port in node.out_ports().values() if not port.disconnected()])
+        if num_of_outputs >= 2:
+            node.out_port(1).data.set_shape(int64_array([max_number_of_boxes, 3]))
+        if num_of_outputs >= 3:
```

Comment:
Check that output port exists. The same for line 92

Reply:
Fixed.